solutionbox/structured_data/datalab_solutions/structured_data/preprocess/cloud_preprocess.py

```diff
@@ -0,0 +1,305 @@
+# Copyright 2017 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import argparse
+import os
+import sys
+import json
+
+import google.cloud.ml as ml
+
+
+INPUT_FEATURES_FILE = 'input_features.json'
+SCHEMA_FILE = 'schema.json'
+
+NUMERICAL_ANALYSIS = 'numerical_analysis.json'
+CATEGORICAL_ANALYSIS = 'vocab_%s.csv'
+
+
+def parse_arguments(argv):
+  """Parse command line arguments.
+
+  Args:
+    argv: list of command line arguments, includeing programe name.
+
+  Returns:
+    An argparse Namespace object.
+  """
+  parser = argparse.ArgumentParser(
+      description='Runs Preprocessing on structured data.')
+  parser.add_argument('--output_dir',
+                      type=str,
+                      required=True,
+                      help='Google Cloud Storage which to place outputs.')
+  parser.add_argument('--input_feature_types',
+                      type=str,
+                      required=True,
```

Comment:
I'll hold off on making things optional for now. Feature types also encodes the default values for training. Picking default values for training is not something I want to do, especially for categorical columns. I want to revisit this late.